src/route/routesSSRData.js

```diff
@@ -214,7 +218,14 @@ export const playlistCollectionPage = (store, match) => {
   return store.sagaMiddleWare.run(mdbSagas.fetchCollection, mdbActions.fetchCollection(cID)).done
     .then(() => {
       const [c] = mdbGetCollectionByIdSelector(store.getState(), [cID]);
+      if (!cuId && !c?.cuIDs) {
+        // This can happen when collection is empty (unit is not secure or published).
+        console.error(`Failed fetching unit for ${cuId} for ${cID}`);
+        return Promise.reject();
+      }
+
       store.dispatch(mdbActions.fetchUnit(cuId || c?.cuIDs[0]));
+      return Promise.resolve(null);
```

Comment:
you dont need return Promise.resolve .then wrap to promise automate what you return

Reply:
I have to as I have return previously so Lint fails on arrow function should have a return value

Reply:
you can return null, but as I understand that you want error, so its good

Reply:
Yes.